src/ctia/domain/id.clj

```diff
@@ -70,13 +73,14 @@
 
 (defn short-id->id
   [type short-id {:keys [hostname path-prefix port protocol]}]
-  (map->CtiaId
-   {:hostname hostname
-    :short-id short-id
-    :path-prefix (not-empty path-prefix)
-    :port port
-    :protocol protocol
-    :type (name type)}))
+  (if (valid-short-id? short-id)
```

Comment:
@yogsototh it'd be interesting to hear why you might prefer `when` over `if` in this spot? This is one of my top five bikeshedding discussion subjects :D /cc @polygloton